Create chapter 2 content

rorywhite200 opened the pull request:
close #55

rorywhite200 committed:
add datasets and data processing for chapter 2 intro exercise

rorywhite200 committed:
add first section of chapter 2 and update yaml

rorywhite200 committed:
Merge remote-tracking branch 'origin/main' into chapter-2-draft

rorywhite200 committed:
add simpson's paradox, grouping details,  color section and shape viz

rorywhite200 committed:
add chapter 2 to yaml

rorywhite200 committed:
rename chapter and change import in yaml

rorywhite200 committed:
Update shape section of chapter 2

rorywhite200 committed:
Convert color section from bullets to paragraphs

joelostblom commented on textbook/2_grouping.qmd:
````diff
@@ -0,0 +1,328 @@
+
+Simpson reported a curious phenomenon where a trend present in several different groups of data disappears when they are combined. 
+
+Below is an illustration of the Simpson's Paradox at play. Notice how the overall dataset does not have a clear direction but when you color by group, both groups exhibit the same negative trend. 
+
+::: {.panel-tabset}
+
+### Overall
+
+```{python}
+simpsons_df = pd.read_csv("data/simpsons_paradox_data_l.csv")
+alt.Chart(simpsons_df).mark_circle().encode(
+  x=alt.X('X'),
+  y=alt.Y('Y'),
+)
+```
````
Do we need to switch to a toy dataset here? Could we continue with the same data set, a smaller version of that data, or a related data set?

joelostblom commented on textbook/2_grouping.qmd:
````diff
@@ -0,0 +1,328 @@
+# Altair
+::: {.panel-tabset}
+
+# Color
+
+```python
+.encode(color = 'category_column')
+```
+
+# Shape
+
+```python
+.encode(shape = 'category_column')
+```
+
+# Size
+
+```python
+.encode(size = 'category_column')
+```
+
+# Position
+
+```python
+.encode(x = 'category_column')
+```
+
+```python
+.encode(y = 'category_column')
+```
+:::
````
No spaces around parameter assignment in python

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,328 @@
+
+## Strategies for grouping
+
+The grammar of graphics gives us a convenient syntax for grouping. We simply define the encoding channel that should represent a categorical variable, using `aes()` in ggplot and `.encode()` in Altair. Color, shape, size and position are the most effective channels for grouping, and they can be easily interchanged:
```
Let's try to keep all mentions of library specific (such as `aes` and `encode`) inside the panels of the respective package

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,328 @@
+
+The main challenge in applying grouping is understanding which encoding channel to use to facilitate the comparisons we want to make. Each grouping strategy has unique strengths and pitfalls, which we shall discuss in detail.
+
+### Color
```
For some reason, this doesn't quite  match what I see in the rendered version. Maybe the render didn't update after you made another commit? 

![image](https://github.com/user-attachments/assets/9f90eee0-e9ad-485b-9cbd-a405dfa7ee18)

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,328 @@
+It is critical that each color be clearly distinct from the others, as too-similar colors can confuse the viewer and make differentiation difficult. Similarly, it is wise to limit the number of distinct categories represented by color: adding too many different colors overwhelms the visualization and makes them harder to distinguish.
+
+Accessibility is another key concern when using color. Colorblindness affects approximately 8% of men and 0.5% of women globally, with the most common variant being deuteranopia, which impairs the ability to distinguish between red and green hues. While Altair’s default color palette is designed to be color-blind friendly, ggplot’s is not.
+
```
```suggestion

We will learn much more about colors later in chapter ...including specifics around which color are safe to use together and which are suitable to represent different data types.
```

rorywhite200 replied:
have added this

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,328 @@
+alt.Chart(simpsons_df).mark_circle().encode(
+  shape='Group',
+  color='Group',
+  x=alt.X('X'),
+  y=alt.Y('Y'),
```
No need to use alt.X/Y when not passing options

joelostblom commented on textbook/2_grouping.qmd:
````diff
@@ -0,0 +1,328 @@
+```{r}
+# Load necessary libraries
+library(ggplot2)
+simpsons_df <- read.csv("data/simpsons_paradox_data_l.csv")
````
let's use `readr::read_csv` over `read.csv`, there are some cases where the latter causes issues that are hard to detect. In general, I wonder if we should read from the public URL here instead of the relative path, so that student can easily copy the code and run on their computers?

rorywhite200 replied:
have changed!

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,328 @@
+
+:::
+
+The combination of shape and color can make grouping clearer than either of these categories alone. However, there are some important caveats. At line-drawings or outlines can make color difficult to distinguish as it reduces the area that is colored. It is preferable to use filled shapes when combining these encoding channels. 
```
What does "At line-drawings" mean?

rorywhite200 replied:
have corrected

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,328 @@
+
+The combination of shape and color can make grouping clearer than either of these categories alone. However, there are some important caveats. At line-drawings or outlines can make color difficult to distinguish as it reduces the area that is colored. It is preferable to use filled shapes when combining these encoding channels. 
+
+Another important consideration is the size of the shapes. It is much easier for the viewer to distinguish between larger shapes than smaller ones. If you have a scatterplot with tiny marks, you may want to restrict the number of unique shapes you show, or consider avoiding shape altogether.
```
It would be good if we can give heuristics for the number of shapes. I'm not sure if the rules are the same as color here, my sense is that it might be even lower

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,328 @@
+
+### Size
+
+Size is more useful for ordinal categories. Hello.
```
hi

joelostblom commented on textbook/img/grouping-strategies.gif:
I don't see this one references anywhere in the doc

rorywhite200 replied:
Yep I decided not to use this. Will delete

joelostblom commented on textbook/data_processing/alcohol-lifespan.py:
Highly appreciated that you include the raw data and the wrangling script! Much better than manually editing and just uploading the cleaned data.

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,328 @@
+::: {.callout-note title="Learning outcomes"}
+- **Explain** how grouping allows you to directly compare categories and helps you to avoid making incorrectly generalized conclusions (e.g., Simpson’s Paradox).
+
+- **Apply** grouping strategies using color, size, shape and position to visually distinguish between categories via `.encode(color='category')` and `aes(color=category)`.
```
One thought I had for LOs in general, is that how do we include code if more than two packages are used in the book... although that can be a future problem as I'm not sure how likely it is to happen and I don't think it is worth removing  the code just for that reason

joelostblom reviewed (commented):
Looking great overall @rorywhite200 ! I have left some comments in addition to our discussion today. I think one of the biggest things left to decide is what to do with the data not starting from 0

rorywhite200 committed:
update opening exercise to use alternative data and approach

rorywhite200 commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,328 @@
+
+In Viridia, historians have been studying datasets from Earth to identify what went wrong and avoid repeating the mistakes of their ancestors.
+
+During their research, they uncovered records of a toxic beverage known as “alcohol.” 
```
Great idea. I have tried to implement this idea in the latest commit by hiding the chart in a hint and asking:

"Try to imagine what this chart could look like. What trend would you expect to see in this data?"

rorywhite200 commented on textbook/2_grouping.qmd:
````diff
@@ -0,0 +1,328 @@
+
+```{python}
+#| label: fig-polar
+#| fig-cap: "Source: Global Burden of Disease Study 2021 (GBD 2021) Results; Human Mortality Database (2023), United Nations - World Population Prospects (2022) – processed by Our World in Data"
````
I tried using footnotes for now as you can hover over them and see the source in place. I should probably add the url too so they can directly download the csv.

rorywhite200 commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,328 @@
+life_expectancy_df = pd.read_csv("data/life_expectancy_alcohol.csv")
+
+alt.Chart(life_expectancy_df).mark_circle().encode(
+    x=alt.X('average_life_expectancy', scale=alt.Scale(zero=False)),
```
Yes I agree, it looks quite weird without fitting to data range. I've switched it out for a dataset showing life expectancy from age 50 which doesn't require using Alt and has the same effect. I tried a bunch of other datasets for this example (deaths, cancer, heart disease, car crashes etc) and it's the best I could find.

rorywhite200 committed:
add mental wellbeing data and image

rorywhite200 committed:
add 50+ life expectancy data from IHME

rorywhite200 committed:
update life expectancy script to merge with income data, add csv and raw data

rorywhite200 committed:
add income to initial exercise, add position section, update visualization examples and add first draft of final mission

rorywhite200 committed:
implement Joel's suggestions

rorywhite200 committed:
refine text

rorywhite200 committed:
adjust wording and add faceting

rorywhite200 committed:
add jitterbug image

rorywhite200 committed:
add position exercise, faceting exercise and update text

rorywhite200 committed:
add raw data, script and processed data for mental wellbeing

rorywhite200 committed:
tidy chapter and improve mission exercise

rorywhite200 committed:
improve ggplot formatting and add ggplot solutions for mission

joelostblom commented on textbook/2_grouping.qmd:
````diff
@@ -0,0 +1,994 @@
+
+# ggplot
+
+In ggplot2, you can achieve the same effect with jittering by using the `position_jitter()` function. Jitter is directly applied to the marks through the position argument within `geom_point()`.
+
+```{r fig.width=8, fig.height=6}
+ggplot(income_df, aes(
+    x=percentage_heavy_drinkers, 
+    y=region, 
+    color=region
+)) + 
+geom_point(position = position_jitter(width = 0, height = 0.2))
````
```suggestion
In ggplot2, you can jitter points by using the `geom_jitter`, which is a combination of `geom_point` together with a function called `position_jitter`.

```{r fig.width=8, fig.height=6}
ggplot(income_df, aes(
    x=percentage_heavy_drinkers, 
    y=region, 
    color=region
)) + 
geom_jitter()
```

Was there a specific reason you opted for position_jitter here? Maybe because it is more explicit? I see geom_jitter more commonly, but happy to switch if you think the other approach is more instructive.

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+:::
+
+
+::: {.callout-caution}
+Jittering can be confusing! It can easily be mistaken for a scatterplot with true variation in the underlying data.
+:::
```
Hmm is this true? Maybe something to discuss.

rorywhite200 replied:
Maybe this is just my problem. Take that chart I shared before. For me, on first glance it looks like the y-axis position of the points has some sort of significance or meaning. 

![image18-1-1024x645](https://github.com/user-attachments/assets/aa3ad4bd-f3c9-4caf-a25c-9cd5f5c692a5)

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+Jittering can be confusing! It can easily be mistaken for a scatterplot with true variation in the underlying data.
+:::
+
+## Faceting
```
```suggestion
### Faceting
```

I think this should still be part of the "grouping strategies" section

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+
+## Faceting
+
+Faceting takes the concept of grouping to the next level by generating multiple small charts, each dedicated to a subset of the data. It is particularly effective for highlighting distinct trends and patterns in each subset, as well as identifying extreme values or outliers within groups. And unlike grouping by position, faceting keeps both the x and y axis free for other variables.
+
+Let's check out the syntax:
```
```suggestion
Faceting takes the concept of grouping to the next level. Instead of using different visual elements within a single chart, it generates multiple small charts, each dedicated to a grouped subset of the data. Faceting is particularly effective for highlighting distinct trends and patterns in each subset, as well as identifying extreme values or outliers within groups. We could think of faceting as a type of position grouping, but its advantage is that it keeps both the x and y axis free for other variables.

Let's check out the syntax:
```

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+
+# Altair
+
+In Altair, faceting by World Bank income group is straightforward. We simply append `.facet('income_group')` to our chart. Additionally, we can control the layout by setting the number of columns using the `columns` argument:
```
```suggestion
To facet by income group, we can append `.facet('income_group')` to our chart. When we are faceting the xy-axes are the same for all the grouped facets/subcharts. The default layout in Altair is to have single row with all the charts spread out horizontally. This would make it easy to compare the charts' y-axis, since they would all be aligned. If we instead are the most interested in comparing the values on the x-axis, we could spread the charts out vertically in a single column. Here, we don't have an explicit primary comparison, but we show how to change from the default horizontal layout by using the `columns` parameter.
```

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+    y='life_expectancy_at_50',
+).facet(
+    'income_group',
+    columns=2
```
```suggestion
    columns=1
```

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+    y = life_expectancy_at_50
+)) +
+geom_point() +
+facet_wrap(~ income_group, ncol=2)
```
````suggestion
facet_wrap(~ income_group, ncol=1)
```

In addition to faceting on a single variable in the data, faceting can also be done over two variables: one spread horizontally and one spread out vertically in the faceted charts. In ggplot, we would swap out `facet_wrap` to `facet_grid` in order to do this. For example, to have facets that are grouped by both income group and region, we could use `facet_grid(vars(income_group), vars(region))`. This can be effective when we need to group our data further, but it does risk making the chart hard to overview, so it is best used when there are fewer categories in the faceting variables.
````

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+
+# ggplot
+
+Facetting works via `facet_wrap`. By default it tries to create an even num of cols and rows, rather than putting all plot in one row like Altair. However, you can specify the desired number of columns with the `ncol` argument.
```
```suggestion
To facet by income group, we can append `.facet_wrap(vars(income_group))` to our chart. The `vars` function is required for `facet_wrap` to understand that we are referring to column names just like `aes` is required when typing out column names for the different visual channels.

When we are faceting the xy-axes are the same for all the grouped facets/subcharts. The default layout in ggplot is to achieve and even number of columns and rows. This is good when we are low on space, but otherwise it is preferred to spread the charts out horizontally in a single row if the primary purpose was to compare the charts' y-axis, since they would all be aligned, or vertically in a single column if we are the most interested in comparing the groups in terms of the values on the x-axis. Here, we don't have an explicit primary comparison, but we show how to change from the default horizontal layout by using the `ncol` parameter.
```

`vars` is recommend over ~ in general nowadays, neither is very intuitive in my opionion.

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+
+:::
+
+Faceting makes it easy to see the overall shape of the distribution for each income group and how tightly clustered the values are. It also reveals outliers in the Upper Middle Income facet – we can easily spot two countries with a higher percentage of heavy drinkers.
```
```suggestion
As you can see above, faceting makes it easy to observse the overall shape of the distribution for each income group and how tightly clustered the values are. It also reveals potential outliers in the Upper Middle Income facet – we can easily spot two countries with a higher percentage of heavy drinkers. On the other hand, it is now harder to compare the exact values between two groups because the points are spread out in different charts, instead of right next to each other in a single chart.
```

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+
+Faceting makes it easy to see the overall shape of the distribution for each income group and how tightly clustered the values are. It also reveals outliers in the Upper Middle Income facet – we can easily spot two countries with a higher percentage of heavy drinkers.
+
+Another advantage is that we can apply additional layers of grouping within each facet. Now its your turn to try this!
```
```suggestion
All the previous grouping strategies that we have discussed can be utilized within each facet as additional layers of grouping. This nested grouping can be powerful to stratify our data according to multiple categorical variables, but keep in mind that it can become confusing if we try to fit too much into a single visualization. Exactly how many levels of grouping you can use depends on how  well separated your data is and how many observations you have; more observations and less separation between groups means that you will have to be more careful and use less levels of grouping to keep the visualization easy to interpret.

Let's try out an example of how we can use faceting together with other grouping strategies.
```

joelostblom commented on textbook/2_grouping.qmd:
````diff
@@ -0,0 +1,994 @@
+    'income_group',
+    columns=2
+)
+```
````
````suggestion
```

In addition to faceting on a single variable in the data, faceting can also be done over two variables: one spread horizontally and one spread out vertically in the faceted charts. In Altair, we would use the `column` and `row` parameter in order to achieve this. For example, to have facets that are grouped by both income group and region, we could use `facet(column='income_group', row='region')`. This can be effective when we need to group our data further, but it does risk making the chart hard to overview, so it is best used when there are fewer categories in the faceting variables.
````

joelostblom commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+
+
+:::::
+
+So where does faceting fall short? For one, presenting each group on a separate chart makes it difficult to estimate the overlap between distributions. Its also harder to compare individual values across facets. 
```
```suggestion
```

I moved this earlier

joelostblom reviewed (changes requested):
Awesome work overall @rorywhite200! I really like how this chapter is shaping up and it will be an important improvement to how the material is currently taught I think. I have made quite a few comments with suggestions, maybe I should have made commits instead that you could review, but hopefully this works ok. I only have the Mission section left, but don't want to lose all of these, so I am submitting them now. Once you have had a read and tried implement those that make sense to you, we can have another call and discuss any disagreements or final polish.

rorywhite200 committed:
Apply suggestions from code review

expand learning outcomes for chapter 2

Co-authored-by: Joel Ostblom <joelostblom@users.noreply.github.com>

rorywhite200 committed:
Update wording of initial teaser exercise

Co-authored-by: Joel Ostblom <joelostblom@users.noreply.github.com>

rorywhite200 committed:
Wording updates, code revisions from Joel's suggestions

Co-authored-by: Joel Ostblom <joelostblom@users.noreply.github.com>

rorywhite200 committed:
Fix YAML cell errors and block placement

rorywhite200 commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+
+:::::
+
+For all of these benefits, there are some drawbacks of grouping by position. For one, it occupies an axis that could have represented a continuous variable. This reduces the information density of your visualization. More worryingly, visually separating each group by position can hide overall trends in the dataset. It also requires significant space.
```
Yes I meant the latter.

Have changed to:
More worryingly, while other grouping methods, such as color, allow you visualize both overall and group-level trends,  overall trends are difficult to see when grouping by position

rorywhite200 commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+
+In Altair, we create a random calculated variable called `jitter` by appending the `.transform_calculate()` method to our chart. We use our newly created `jitter` variable, encoding it as the `yOffset` of each data point using `yOffset='jitter:Q'`. This shifts the `yOffset` of each point by a random value. 
+
+The `:Q` simply tells Altair that `jitter` is a continuous quantitative variable.
```
Agreed! it seems like a basic functionality that is more difficult than it needs to be

rorywhite200 committed:
Change life expectancy data to at age 65 (retirement) and include footnote

rorywhite200 commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+:::
+
+In this case, grouping makes the insidious Simpson's Paradox visually unmistakable. This is a prime example of the dual purpose of data visualization. It doesn't just present our data to others; it deepens our understanding of it. 
+
```
That makes sense! I did this and moved the text "While position and color are generally the most effective grouping strategies, it is important to understand the unique strengths and pitfalls of each strategy to chose the one that best supports the comparisons we want to make. This is what we will learn about next" to before the grouping strategies section.

rorywhite200 committed:
add initial grouping viz to demonstrate comparing values

rorywhite200 committed:
add ref for color as preattentive feature

rorywhite200 commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+
+Conveniently, color doesn't alter the layout or structure of the data visualization, so it's an efficient way to compare the exact values of data points between groups.
+
+Let's use color to group our life expectancy data by world region:
```
Have added the ref!

joelostblom replied:
Intriguing question setup in the paper!

> In some cases, including that search for circles among squares, the reaction time (RT) to declare that the target was present or to declare that all the items were the same did not vary as a function of how many items were presented. The slope of the RT x set size function was near zero. This strongly suggested that the all the items were being processed “in parallel” [[2](https://pmc.ncbi.nlm.nih.gov/articles/PMC6513732/#R2)]. In other cases, the RT increased as a roughly linear function of set size, suggesting a serial process [though that need not be the case [3](https://pmc.ncbi.nlm.nih.gov/articles/PMC6513732/#R3),[4](https://pmc.ncbi.nlm.nih.gov/articles/PMC6513732/#R4)]. What made some targets “pop-out”?

rorywhite200 committed:
move position exercise below jittering and change bridging paragraph

rorywhite200 commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+
+Notice how easy it is to compare the distributions between regions?
+
+We can even take it one step further. Position can be combined with color to reinforce grouping on one variable or to introduce grouping on a second variable. 
```
Yep I like this idea. I've moved the section below jittering. The chart is a lot more interpretable with jittering, otherwise it seems kinda pointless to group by color.

rorywhite200 commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+
+::::: {.callout-default .ex-prompt}
+
+You are trying to understand how life expectancy varies across income groups globally, and the role world region plays in these differences.
```
I tried to do this subtly without distracting from the main question

rorywhite200 committed:
Add reference for World Bank and remove old references to life expectancy (all now comes from GBD 2021)

rorywhite200 commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+
+::::
+
+:::::
```
@joelostblom I like this it feels like a proper story. Also makes sense why you are in such high demand in all of these Viridian departments!

rorywhite200 commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+
+There is also the challenge of accessibility. Colorblindness affects approximately 8% of men and 0.5% of women globally. The most common variant is deuteranopia. This impairs the ability to distinguish between red and green hues, so designing your data visualization with this in mind is essential. While Altair’s default color palette is designed to be color-blind friendly, ggplot’s is not.
+
+We will learn much more about accessibility in chapter ...including specifics around which colors are safe to use together and which are suitable to represent different data types.
```
@joelostblom Which comment do you mean?

rorywhite200 commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+
+Now our visualizations are really taking shape! Shape can be combined with color to provide an additional visual cue. Let's take a look at a classic example: the infamous traffic light. 
+
+Do you think that traffic lights are well designed? At first glance,  they break a major rules of accessibility, using the worst possible colors for people with deuteranopia (red and green). Thankfully, in many countries, traffic lights are encoded using both color and shape. A third encoding channel - position - is then used to establish predictable y-axis positions for both 'Stop' and 'Go' sign ...
```
@joelostblom  page 101 of Visualization Analysis & Design by Tamara Munzner could be a good shout. It has the hierarchy of categorical encoding channels, where shape is actually bottom.

rorywhite200 commented on textbook/2_grouping.qmd:
```diff
@@ -0,0 +1,994 @@
+
+While this combined approach makes grouping clearer, there are some pitfalls. For instance, line icons limit the total area that can be colored, reducing the impact of color. It is better to use solid shapes when combining shape and color.
+
+Additionally, size matters more than you'd think. Smaller shapes are difficult to distinguish. When plotting a dense scatterplot of small marks, try restricting the number of unique shapes to three or fewer, or consider avoiding using shape altogether.
```
@joelostblom  I think I made this rule up... Visualization Analysis & Design by Tamara Munzner says that with 10x10 pixels you can make around 12 distinguishable shapes. Grouping by 12 shapes seems it could be a spaghetti nightmare

joelostblom committed:
Be explicit about that the first box has a chapter summary

joelostblom committed:
Format points to avoid redundant spacing

joelostblom committed:
Improve language and flow

joelostblom committed:
Hide code consistently

joelostblom committed:
Improve language and flow

joelostblom committed:
Make R more silent

While these can be useful headsup at time,
they are mostly distracting when teaching other topics
such as viz.

joelostblom committed:
Improve language and flow

joelostblom committed:
Use more flexible aes and facet syntax

`+ aes` clarifies the ability to change aes for an existing chart.
`vars` is preferred of `~` when faceting
(it's the equivalent to aes in a faceting context)

joelostblom committed:
Fix typo

joelostblom committed:
Improve language and flow

joelostblom committed:
Add example of narrowing jittered points

rorywhite200 committed:
add file with simpsons paradox data processing

rorywhite200 committed:
Merge branch 'chapter-2-draft' of github.com:joelostblom/viz-oer into chapter-2-draft

joelostblom commented:
Merging this as is per our discussion and we can follow up on the last exercise changes separately.
